sanityv3/schemas/objects/textTeaser.tsx

```diff
@@ -0,0 +1,142 @@
+import { RadioIconSelector } from '../components'
+import CompactBlockEditor from '../components/CompactBlockEditor'
+import { configureBlockContent, configureTitleBlockContent } from '../editors'
+import { validateCharCounterEditor } from '../validations/validateCharCounterEditor'
+
+import type { PortableTextBlock, Rule } from 'sanity'
+import type { DownloadableImage } from './downloadableImage'
+import type { DownloadableFile } from './files'
+import type { LinkSelector } from './linkSelector'
+import { ThemeSelectorValue } from '../components/ThemeSelector'
+
+const titleContentType = configureTitleBlockContent({
+  highlight: true,
+  styles: [
+    {
+      title: 'Normal',
+      value: 'normal',
+    },
+  ],
+})
+
+const titleAlignmentOptions = [
+  { value: 'left', icon: LeftAlignedImage },
+  { value: 'right', icon: RightAlignedImage },
+]
+
+const blockConfig = {
+  h1: false,
+  h2: false,
+  h3: false,
+  h4: false,
+  internalLink: false,
+  externalLink: false,
+  attachment: false,
+  lists: true,
+}
+
+const blockContentType = configureBlockContent({ ...blockConfig })
+
+export type TextTeaser = {
+  _type: 'textTeaser'
+  title?: PortableTextBlock[]
+  text?: PortableTextBlock[]
+  action?: (LinkSelector | DownloadableFile | DownloadableImage)[]
+  titlePosition?: string
+  theme?: ThemeSelectorValue
+}
+
+export default {
+  name: 'textTeaser',
+  title: 'Text Teaser',
+  type: 'object',
+  localize: true,
+  fieldsets: [
+    {
+      name: 'link',
+      title: 'Link',
+      description: 'Select either an internal link or external URL.',
+    },
+    {
+      name: 'design',
+      title: 'Design options',
+    },
+  ],
+  fields: [
+    {
+      name: 'title',
+      type: 'array',
+      components: {
+        input: CompactBlockEditor,
+      },
+      of: [titleContentType],
+      validation: (Rule: Rule) => Rule.required(),
+    },
+    {
+      name: 'text',
+      title: 'Text content',
+      type: 'array',
+      of: [blockContentType],
+      validation: (Rule: Rule) =>
+        Rule.custom((value: PortableTextBlock[]) => {
+          return validateCharCounterEditor(value, 600)
+        }).warning(),
+    },
+
+    {
+      name: 'action',
+      title: 'Link/action',
+      description: 'Select the link or downloadable file for the teaser',
+      type: 'array',
+      of: [
+        { type: 'linkSelector', title: 'Link' },
+        { type: 'downloadableImage', title: 'Downloadable image' },
+        { type: 'downloadableFile', title: 'Downloadable file' },
+      ],
+      validation: (Rule: Rule) => Rule.max(1).error('Only one action is permitted'),
+    },
+    {
+      name: 'titlePosition',
+      title: 'Title position',
+      description: 'Select which side of the teaser the title should be displayed at on larger screens.',
+      type: 'string',
+      fieldset: 'design',
+      components: {
+        input: function ({ onChange, value }: { onChange: any; value: string }) {
```

Comment:
any on sanityv3 is bad... we should also refactor the RadioIconSelector...

`onChange: (event: PatchEvent) => void`